cdap-app-templates/cdap-etl/cdap-data-pipeline/src/main/java/io/cdap/cdap/datapipeline/draft/DraftRequest.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ */
+
+package io.cdap.cdap.datapipeline.draft;
+
+
+import io.cdap.cdap.api.artifact.ArtifactSummary;
+
+import java.util.Objects;
+
+/**
+ * Request to store a draft.
+ *
+ * @param <T> Type of config that this DraftRequest contains
+ */
+public class DraftRequest<T> {
+  private final String previousHash; // currently unused but will be used later to detect collision
+  private final String name;
+  private final String description;
+  private final int revision; // currently unused but will be used later to enable draft version tracking
+  private final ArtifactSummary artifact;
+  private T config;
+
+  public DraftRequest(T config, String previousHash, String name, String description, int revision,
+                      ArtifactSummary artifact) {
+    this.config = config;
+    this.previousHash = previousHash;
+    this.name = name;
+    this.description = description;
+    this.revision = revision;
+    this.artifact = artifact;
+  }
+
+  public String getDescription() {
+    return description;
+  }
+
+  public String getName() {
+    return name == null ? "" : name;
+  }
+
+  public int getRevision() {
+    return revision;
+  }
+
+  public ArtifactSummary getArtifact() {
+    return artifact;
+  }
+
+
+  public T getConfig() {
+    return config;
+  }
+
+  public String getPreviousHash() {
```

Comment:
The previousHash will be used later on to detect collision in drafts. It represents the hashcode that was provided by the backend when the UI loaded this draft. When the UI makes a save request it will pass that hashcode and the backend can check against the current hash to see if the draft has been modified since the user loaded the draft. 

I added these fields now (even though their not needed yet) so we can avoid changing the table/object schema

Reply:
You probably want to add some javadoc to it.

Reply:
done